### DISCUSSION TITLE
Deliver could verify the screenshot extension before uploading

### DISCUSSION THREAD

**Discussion by JaviSoto** [fastlane/fastlane#17191] [Ideas]
This could be related to https://github.com/fastlane/fastlane/issues/17139

We were having issues uploading our screenshots. We're still investigating, but it seems at least one issue may have been that some of our screenshots were JPEGs, but had .png file extensions.

Renaming them allowed us to finally upload them, so it seems it could be useful for fastlane to verify this and show a clear message to the user if the extension doesn't match, as it could save some people a lot of time!

**Comment by ainame** [comment]
Hey, I'm working on improving screenshots uploads recently. I love your idea. I personally ran into this issue several times also. 
Checking file type can be easily done with the gem fastlane already depends on. It's feasible to implement. 
https://github.com/sdsykes/fastimage 

>  it could save some people a lot of time!

I agree! I will pick this up and try to implement near future😄

**Reply by ainame** [reply]
I read through `Deliver::AppScreenshot.is_valid?` and realised that existing error handling in its validation is not implemented well...
https://github.com/fastlane/fastlane/blob/7add7e9432c351a144809023013bb7ff6c529daf/deliver/lib/deliver/app_screenshot.rb#L86-L94

I would start this from refactoring existing image loading flow. For example, if a screenshot's size isn't supported, then fastlane will crash with an error message for the single image, not for assessing all the images. I would want to know all the issues there at once.

Oddly,  L93 in `initialize` tries to check if an image is valid and just output error message if it's wrong. However, L349 will crash in that case, which doesn't make any sense...
https://github.com/fastlane/fastlane/blob/7add7e9432c351a144809023013bb7ff6c529daf/deliver/lib/deliver/app_screenshot.rb#L93

https://github.com/fastlane/fastlane/blob/7add7e9432c351a144809023013bb7ff6c529daf/deliver/lib/deliver/app_screenshot.rb#L333-L351

**Comment by ainame** [comment]
I have finally implemented this here! Let's see how it goes...

https://github.com/fastlane/fastlane/pull/17708
